src/Engine/movement_handler.py

```diff
@@ -10,21 +12,34 @@ def move_player(player: Player, key_depressed, movement_distance_x, movement_dis
     right = 100  # 'd' key ascii
     down = 115  # 's' key ascii
 
+    up_image = asset_dict.get_asset("frog")
+    up_image2 = asset_dict.get_asset("frog_jumping")
+    left_image = pygame.transform.rotate(up_image, 90)
+    left_image2 = pygame.transform.rotate(up_image2, 90)
+    right_image = pygame.transform.rotate(up_image, -90)
+    right_image2 = pygame.transform.rotate(up_image2, -90)
+    down_image = pygame.transform.rotate(up_image, 180)
+    down_image2 = pygame.transform.rotate(up_image2, 180)
```

Comment:
I think it would be better to put these variable declarations and rotations in the if branches. This function runs on every frame so it's better not to allocate this memory until we know it'll be needed.

Reply:
changes have been made